Make outputType recognize calls to built-in types

tekknolagi opened the pull request:
Calling `str(x)` (where `str` is still the built-in type) should always return `UnicodeExact`. Use the built-in type map and `TBuiltinExact` to filter out subclasses, because subclasses can do whatever they want.

tekknolagi committed:
Make outputType recognize calls to built-in types

Calling `str(x)` (where `str` is still the built-in type) should always
return `UnicodeExact`. Use the built-in type map and `TBuiltinExact` to
filter out subclasses, because subclasses can do whatever they want.

carljm commented:
Internally this is failing tests in `test_cinderjit` and `test_perf_profiler` when run with JIT enabled. Can you repro?

tekknolagi commented:
Huh. I don't have a CI repro -- looks fine to me on my actions: https://github.com/tekknolagi/cinder/actions/runs/5115083456/jobs/9195993151

I can try running locally later.

Can you provide me with some more detail? What is the exact test invocation? Is it crashing? ...?

tekknolagi commented:
It looks like `test_unwatch_builtins` is interfering with the `LoadMethodModule` tests for some reason. If that runs first, the latter fail (everything else was commented out). If `test_unwatch_builtins` is run in a subprocess the issue goes away. I haven't looked harder yet.

tekknolagi commented:
What I don't understand is if I limit the JIT list to functions only present in the LoadModuleMethod tests and insert a hardware breakpoint inside my SSA change (which never fires), the tests still fail. So I'm not changing the codegen, the codegen is very limited in scope, and still, test failure.

tekknolagi commented:
Oh actually if I comment out my changes entirely the tests still fail. So I think that test just needs to be run in a subprocess.

tekknolagi committed:
Run test_unwatch_builtins in subprocess

This test has global effects that affect other tests. Run it in a
subprocess for isolation.

tekknolagi commented:
Augh, hold on, sorry
